testing/regress/regress.md

```diff
@@ -0,0 +1,233 @@
+# Regression Testing
+
+## Why Regression Testing is Important
+
+Adding new features and making improvements is crucial for any product's evolution. However, these changes can sometimes inadvertently introduce regressions – previously fixed bugs that resurface or unexpected issues in existing functionality.
+
+The Regression Suite acts as a safety net, systematically verifying that core application functionality continues to work as expected after modifications. By running these tests, we can proactively identify and address regressions early in the development cycle, preventing them from reaching users and ensuring a stable experience. Think of it as a comprehensive health check for our software after any significant change.
+
+For official releases, the regression suite runs automatically as part of the build cycle. However, individual developers can (and should) use it when making significant changes, before issuing a Pull Request.
+
+## When to Use the Regression Suite
+
+Run the Regression Suite at these key points during the software development lifecycle:
```

Comment:
Not suire of the value of this section.  The regression suite should be run after any change to the platform.  It is automatically run as part actions for each PR.  You have already mentioned that above though.